joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala

```diff
@@ -651,47 +1080,102 @@ class AstCreator(filename: String, global: Global)
   def astForInvocationWithParenthesesPrimaryContext(ctx: InvocationWithParenthesesPrimaryContext): Ast = {
     val methodIdAst = astForMethodIdentifierContext(ctx.methodIdentifier())
     val parenAst    = astForArgumentsWithParenthesesContext(ctx.argumentsWithParentheses())
+    val callNode    = methodIdAst.nodes.filter(_.isInstanceOf[NewCall]).head.asInstanceOf[NewCall]
+    callNode.name(getActualMethodName(callNode.name))
 
     if (ctx.block() != null) {
       val blockAst = astForBlockContext(ctx.block())
-      methodIdAst.withChild(parenAst).withChild(blockAst)
+      callAst(callNode, Seq[Ast](parenAst, blockAst))
     } else {
-      methodIdAst.withChild(parenAst)
+      callAst(callNode, Seq[Ast](parenAst))
     }
   }
 
   def astForIsDefinedExpressionContext(ctx: IsDefinedExpressionContext): Ast = {
-    astForExpressionContext(ctx.expression())
+    val exprAst = astForExpressionContext(ctx.expression())
+    val callNode = NewCall()
+      .name(ctx.IS_DEFINED().getText)
+      .code(ctx.IS_DEFINED().getText)
+      .methodFullName(MethodFullNames.OperatorPrefix + ctx.IS_DEFINED().getText)
+      .signature("")
+      .dispatchType(DispatchTypes.STATIC_DISPATCH)
+      .typeFullName(Defines.Any)
+      .lineNumber(ctx.IS_DEFINED().getSymbol().getLine())
+      .columnNumber(ctx.IS_DEFINED().getSymbol().getCharPositionInLine())
+    callAst(callNode, Seq[Ast](exprAst))
   }
 
   def astForIsDefinedPrimaryContext(ctx: IsDefinedPrimaryContext): Ast = {
     astForExpressionOrCommandContext(ctx.expressionOrCommand())
   }
 
   def astForJumpExpressionPrimaryContext(ctx: JumpExpressionPrimaryContext): Ast = {
-    // TODO to be implemented
-    Ast()
+    if (ctx.jumpExpression().RETURN() != null) {
+      val retNode = NewReturn()
+        .code(ctx.getText)
+        .lineNumber(ctx.jumpExpression().RETURN().getSymbol().getLine)
+        .columnNumber(ctx.jumpExpression().RETURN().getSymbol().getCharPositionInLine)
+      returnAst(retNode, Seq[Ast]())
+    } else if (ctx.jumpExpression().BREAK() != null) {
+      val node = NewControlStructure()
```

Comment:
Are there maybe `x2cpg` functions you can use here instead?

Reply:
Did not notice the controlStructureAst() being used in the other front end code whose reference I took. I will change this in another PR since that will result in some more diff